### ISSUE TITLE
 "body"-field does not work inside object-widget

### ISSUE THREAD

**Issue by speexy** [netlify/netlify-cms#1186]
**- Do you want to request a *feature* or report a *bug*?**

Bug? It kind of makes sense that it does not work, but it does feel buggy.

**- What is the current behavior?**

When I do something nested like
```
widget: "object"
fields:
 - label: "Body"
   name: "body"
   widget: "markdown"
```
then this widget doesn't pull in the body from the respective markdown file

**- If the current behavior is a bug, please provide the steps to reproduce.**

use body-field together with the object-widget, see that in the cms the body-field is empty

**- What is the expected behavior?**

I assumed that `name: "body"` wasn't limited to work on only the first layer of fields

*Why would I want to do something like this anyway?*

I was looking for a solution to use anchors/aliases for fields and "merge" them like so:
```
shared_fields: &shared
 - title
 - image

unique_set_of_fields:
 - *shared
 - a unique field
```
I was about to ask for a "flattening"-feature when I came up with:
```
meta: &meta
 - meta field
header: &header
 - title_field
 - image_field
```
and
```
content: &content
 - label: "Header"
   name: "header"
   fields: *header
 - {label: "body", name: "body", widget:"markdown"} 
```
that way I can have anchors and the nesting makes some sense on the cms. I can just include `*content` in moste places, and in the places where the content fields are slightly different I'll compose it from smaller units. Except that I cannot nest the body field. 

Now that I'm writing about it, maybe a flattening-feature would be better after all? Should I open an issue for that?

**Comment by erquhart** [comment]
Can you explain the flattening concept a bit more?

The CMS treats the entire entry as a tree, and the body must be in the top of layer of that tree, as it cannot be output in a nested fashion. Same for the title field, as the CMS uses that for specific roles like slugification.

**Comment by speexy** [comment]
Sure! I'll try :)

Let's say I have:
```
shared: &shared
  - title
  - body
unique:
  - *shared
  - unique 
```
That would return a structure like this:
```
{unique: [[title, body], unique]}
```
by flattening I mean returning this:
```
{unique: [title, body, unique]}
```
This flattening/merging would make it possible to write a cleaner config.yml by using anchors/aliases. I would be ok with the nested results too, however, I am run into problems because then the title and the body are not on the top layer anymore.

What I am looking for is possible with dictionaries:
```
shared: &shared
  title: title
  body: body
unique:
  <<: *shared
  unique: unique 
```

The following is mostly derived from [this stack overflow question:](https://stackoverflow.com/questions/9254178/is-there-yaml-syntax-for-sharing-part-of-a-list-or-map)

It is possible to create dictionaries with null values like this:
```
shared: &shared
  ? title
  ? body
unique:
  <<: *shared
  ? unique 
```
or this:

```
shared: &shared
  title:
  body:
unique:
  <<: *shared
  unique: 
```
maybe dictionaries with null values could be converted to lists, where the dictionary key becomes the list value?

or maybe something along those lines (also from [stackoverflow](https://stackoverflow.com/questions/9254178/is-there-yaml-syntax-for-sharing-part-of-a-list-or-map)):
```
shared: &shared
  - title
  - body
unique:
  MERGE
    - *shared
    - unique 
```
the results should look like this:
```
{
  "unique": [
    "title", 
    "body", 
    "unique"
  ]
}
```

To summarize:
I am looking for  way to organize re-occuring field combinations as variables and use them together with fields that are unique for a specific page. With standard yaml syntax this will lead to nested lists. This leads to problems for the title and body field as they need to be at the top layer. 
Therefore what I am asking is:
- Is it possible to add custom syntax and process the config.yml on your side in such a way that nested lists will be flattened?
- If not, is it possible to change the title and body field in such a way that they can be used anywhere in the tree?

I hope I could make myself understood and it is more clear what I'm referring to with this issue!

**Comment by erquhart** [comment]
Man, YAML can do some crazy stuff 😛 

I definitely understand your goal! I'm just not sure we should bend how things work in the CMS to accommodate, as we're able to currently take the config at face value without interpreting much.

**Comment by speexy** [comment]
I get that. Would be an awesome feature though!
Any chance on the body and title field being usable from anywhere in the tree as an alternative?

**Comment by erquhart** [comment]
@Benaiah and I were discussing how Netlify CMS might provide this sort of thing natively, by allowing fields or named field groups to be defined separately from collections, and then applied wherever needed.

**Comment by mhrubikupgrade** [comment]
Hi @erquhart , there are any updates on this?

**Comment by erquhart** [comment]
No, haven't discussed it since the last comment.

**Comment by andreasvirkus** [comment]
Guess this is linked to https://secure.helpscout.net/conversation/666163225/42?folderId=2338326
(or it would at least help partially solve this).

**Comment by erquhart** [comment]
That's a private link, can't access.

What we're talking about is similar to #1342, which I _think_ would help with the body field issue.

**Comment by andreasvirkus** [comment]
Heh, yes.. I think my clipboard manager was having an off day; that's the ticket I meant 😅

**Comment by erquhart** [comment]
Closing in favor of #1342.

**Comment by ajpalok** [comment]
Hey, I added body field of a article using markdown widget but not showing on preview.
something like 

      - label: "Body"
        name: "body"
        widget: "markdown"

after writing article preview doesn't showing that... what to do now??

**Comment by erezrokah** [comment]
Hi @ajpalok, please see https://www.netlifycms.org/docs/customization/ or open a new issue following the template
